optuna/trial/_trial.py

```diff
@@ -768,29 +757,6 @@ def number(self):
 
         return self.storage.get_trial_number_from_id(self._trial_id)
 
-    @property
-    def trial_id(self):
```

Comment:
[Comment]
`ChainerMNTrial` also has `trial_id`, but I think it is out of the scope of this PR.

Reply:
Thanks for the comment! I agree with your comment that the fix for `integration` module is out of the scope of this PR.

Reply:
Thank you for your fix.

@hvy I prioritized consistency with the title, but it may be better to prioritize consistency with the implementation of the `Trial` class. If you have any concerns, please point it out.

Reply:
This is not a strong opinion as we'll address it soon in any case but yes I'd personally include the change in this PR. I'm fine with either. Thanks for letting me know!

Reply:
Thank you for your reply. I'm also fine with either choice, so please merge after your review.

Reply:
Just leaving a note but here's the ChainerMN follow up https://github.com/optuna/optuna/pull/1375